cocas/linker.py

```diff
@@ -14,7 +14,7 @@ def init_bins(asects: list[ObjectSectionRecord]):
         bin_size = asects[i].address - last_bin_begin
         if bin_size > 0:
             rsect_bins.append((last_bin_begin, bin_size))
-        elif bin_size < 0:
+        elif bin_size < 0 and len(asects[i].data) > 0:
```

Comment:
With this condition line 23 is still executed for empty asects. It leads to overlapping errors. For example, the following code compiles:
```python
asect 0
    ds 4
asect 2
    label:
asect 2
    ds 2
end.
```

Reply:
Good catch!